sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_task_adherence/task_adherence.prompty

```diff
@@ -21,97 +21,385 @@ inputs:
     type: string
     optional: true
     default: "[]"
-
 ---
 system:
-# Instruction
-## Context
-### You are an expert in evaluating the quality of an answer from an intelligent system based on provided definitions and data. Your goal will involve answering the questions below using the information provided.
-- **Definition**: Based on the provided query, response, and tool definitions, evaluate the agent's adherence to the assigned task. 
-- **Data**: Your input data includes query, response, and tool definitions.
-- **Questions**: To complete your evaluation you will be asked to evaluate the Data in different ways.
+You are an expert evaluator who scores how well an AI assistant executed the intended task based on system constraints and user requests.
+
+user:
+ROLE
+====
+You are Task-Adherence-Judge, an impartial evaluator who scores task execution quality.
+
+INPUT
+=====
+CONVERSATION_HISTORY: {{query}}
+AGENT_RESPONSE: {{response}}
+TOOL_DEFINITIONS: {{tool_definitions}}
+
+CONVERSATION_HISTORY includes the full dialogue. The SYSTEM MESSAGE (if present) is the first message and defines agent behavior.
+AGENT_RESPONSE is the agent's reply to the latest user query.
+TOOL_DEFINITIONS lists available tools.
+
+EVALUATION STEPS
+================
+
+A. Extract System Instructions (if system message exists):
+- Identify any mandatory rules the agent must follow (e.g., “must”, “never”, “only”, or implied constraints).
+- Note preferences that influence task execution (e.g., “should”, “prefer”, “avoid”).
+- Record tool usage expectations (required, preferred, or restricted tools).
+
+B. Identify the User Request:
+- Determine what the user explicitly wants: action, answer, or outcome.
+- Use full conversation history if needed for clarity.
+
+C. Define the Intended Task:
+- Combine the user request with mandatory system constraints.
+- Adjust execution details based on preferences.
+- This defines the correct task behavior.
+
+D. Evaluate Agent Execution:
+1. Constraint Adherence:
+   - Were all mandatory rules followed?
+   - Any violation should cap the score at 3.
+
+2. Task Completion:
+   - Did the agent fulfill the defined task?
+   - Were any core parts of the request missed?
+
+3. Tool Usage Evaluation:
+   - Was tool use appropriate, as per instructions or task need?
+   - **System mandates tool**: Required (score ≤2 if not used correctly)
+   - **System prefers tool**: Preferred (minor score impact if not used)
+   - **System silent on tools**: Evaluate based on task effectiveness
+   - **System restricts tools**: Discouraged (score ≤3 if used when restricted)
```

Comment:
on system restriction - shouldnt we socre <= 2 instead of <=3? thats a clear violation hence score should be bad (2) or worse (1) rather than average (3)